### ISSUE TITLE
"kbox provision" errors out due to "VirtualBox is still running" when VB is off.

### ISSUE THREAD

**Issue by reynoldsalec** [kalabox/kalabox-issues#218]
No VirtualBoxes are running, although I still have my Kalabox V1 VB.

Ran "kbox provision", got the following:

```
STEP [13] -- Setting things up....
 ##### ADMINISTRATIVE PASSWORD WILL BE REQUIRED! #####
 --- Installing: /tmp/Boot2Docker-1.4.1.pkg
 --- Setting up DNS for Kalabox.
 - 0:275:
 - execution error: installer: Error - The installer has detected that VirtualBox is still running. Please shutdown all running VirtualBox machines and exit VirtualBox then restart the installation. (1)

 --- Finished installing

STEP [14] -- Setting up and turning on the Kalabox VM....
 --- undefined
STEP [15] -- Installing core services....

/Users/alec/workspace/kalabox/lib/engine.js:24
    ['Provider "' + engine.getProviderName() + '"', msg].join(' ')
                           ^
TypeError: Object #<Object> has no method 'getProviderName'
    at makeProviderError (/Users/alec/workspace/kalabox/lib/engine.js:24:28)
    at /Users/alec/workspace/kalabox/lib/engine.js:37:18
    at /Users/alec/workspace/kalabox/lib/engine/provider/b2d.js:161:7
    at /Users/alec/workspace/kalabox/lib/util/shell.js:36:5
    at /Users/alec/workspace/kalabox/node_modules/shelljs/src/exec.js:107:7
    at ChildProcess.exithandler (child_process.js:651:7)
    at ChildProcess.EventEmitter.emit (events.js:98:17)
    at maybeClose (child_process.js:753:16)
    at Process.ChildProcess._handle.onexit (child_process.js:820:5)
```

Running "kbox provision" again sees the same issue.

**Comment by reynoldsalec** [comment]
Oh, I did run ./scripts/uninstall.sh, but didn't run the "Uninstall VirtualBox" last step.

@pirog any ideas on what's goofing things up?

**Comment by pirog** [comment]
make sure that virtual box is not open as you run the installer

On Tue, Feb 10, 2015 at 3:09 PM, reynoldsalec <notifications@github.com>
wrote:

> Oh, I did run ./scripts/uninstall.sh, but didn't run the "Uninstall
> VirtualBox" last step.
>
> @pirog <https://github.com/pirog> any ideas on what's goofing things up?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox-issues/issues/218#issuecomment-73773922>
> .
>



-- 
Cheers,

Mike Pirog
Kalamuna
www.kalamuna.com

**Comment by pirog** [comment]
may need to have all kboxes off. i would turn them all off and close the app before you try install again. might want to run the uninstall first again as well to make sure we have a clean slate.

**Comment by reynoldsalec** [comment]
VirtualBox was definitely off/not open when running the installer. Running uninstall.sh and choosing to delete VirtualBox fixed the issue.

Interestingly, I had to run the uninstaller twice, the first time it said two VB processes were still running, second time no problems. There may be a couple processes that hang around after all boxes are shutdown and the GUI is closed, could be a drag for some folks. Otherwise, feel like this is a none-issue.

**Comment by bcauldwell** [comment]
What was the fix for this?

On Tue, Feb 10, 2015 at 4:30 PM, reynoldsalec <notifications@github.com>
wrote:

> Closed #218 <https://github.com/kalabox/kalabox-issues/issues/218>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox-issues/issues/218#event-233645880>.
>



-- 
- Benjamin Cauldwell
